mslib/mswms/dataaccess.py

```diff
@@ -213,7 +222,12 @@ def _determine_filename(self, variable, vartype, init_time, valid_time, reload=T
         """
         assert self._filetree is not None, "filetree is None. Forgot to call setup()?"
         try:
-            return self._filetree[vartype][init_time][variable][valid_time]
+            filename = self._filetree[vartype][init_time][variable][valid_time]
+            mtime = os.path.getmtime(os.path.join(self._root_path, filename))
+            if filename in self._file_cache and mtime == self._file_cache[filename][0]:
+                return filename
+            else:
+                raise KeyError
         except KeyError:
```

Comment:
os.path.getmtime can fail because of an OSError when path not exists or inaccessible is. The KeyError can occure only when that is passed.

When the KeyError happened the filename can be removed from self._file_cache